src/widget/sidebar/military.c

```diff
@@ -0,0 +1,732 @@
+
+static image_button buttons_title_close[] = {
+    {127, 5, 31, 20, IB_NORMAL, 90, 4, button_close_military_sidebar, button_none, 0, 0, 1},
+    {4, 3, 117, 31, IB_NORMAL, 93, 0, button_military_menu, button_none, 0, 0, 1}
+};
+
+static arrow_button buttons_cycle_legion[] = {
+    {1, 0, 19, 24, button_cycle_legion, 0, 0},
+    {125, 0, 21, 24, button_cycle_legion, 1, 0},
+};
+
+static generic_button buttons_formation_layout[] = {
+    {0, 0, 29, 29, button_select_formation_layout, button_none, 0, 0},
+    {31, 0, 29, 29, button_select_formation_layout, button_none, 1, 0},
+    {62, 0, 29, 29, button_select_formation_layout, button_none, 2, 0},
+    {93, 0, 29, 29, button_select_formation_layout, button_none, 3, 0},
+    {124, 0, 29, 29, button_select_formation_layout, button_none, 4, 0}
+};
+
+static generic_button buttons_single_legion_bottom[] = {
+    {0, 0, 73, 48, button_go_to_legion, button_none, 0, 0},
+    {77, 0, 73, 48, button_return_to_fort, button_none, 1, 0},
+};
+
+static generic_button buttons_select_legion[] = {
+    {0, 0, SIDEBAR_EXPANDED_WIDTH - 12, 50, button_select_legion, button_none, 0, 0},
+    {0, 50, SIDEBAR_EXPANDED_WIDTH - 12, 50, button_select_legion, button_none, 1, 0},
+    {0, 100, SIDEBAR_EXPANDED_WIDTH - 12, 50, button_select_legion, button_none, 2, 0},
+    {0, 150, SIDEBAR_EXPANDED_WIDTH - 12, 50, button_select_legion, button_none, 3, 0},
+    {0, 200, SIDEBAR_EXPANDED_WIDTH - 12, 50, button_select_legion, button_none, 4, 0},
+    {0, 250, SIDEBAR_EXPANDED_WIDTH - 12, 50, button_select_legion, button_none, 5, 0},
+};
+
+static const int IMAGE_OFFSETS_TO_FORMATION[7] = {
+    FORMATION_COLUMN,
+    FORMATION_TORTOISE,
+    FORMATION_DOUBLE_LINE_1,
+    FORMATION_DOUBLE_LINE_2,
+    FORMATION_MOP_UP,
+    FORMATION_SINGLE_LINE_1,
+    FORMATION_SINGLE_LINE_2
+};
+
+static const int LAYOUT_IMAGE_OFFSETS_LEGIONARY[2][FORMATIONS_PER_LEGION] = {
+    {0, 0, 2, 3, 4}, {0, 0, 3, 2, 4},
+};
+
+static const int LAYOUT_IMAGE_OFFSETS_OTHER[2][FORMATIONS_PER_LEGION] = {
+    {5, 6, 2, 3, 4}, {6, 5, 3, 2, 4},
+};
+
+static const int LAYOUT_BUTTON_INDEXES_LEGIONARY[2][FORMATIONS_PER_LEGION] = {
+    {
+        FORMATION_TORTOISE,
+        FORMATION_COLUMN,
+        FORMATION_DOUBLE_LINE_1,
+        FORMATION_DOUBLE_LINE_2,
+        FORMATION_MOP_UP
+    },
+    {
+        FORMATION_TORTOISE,
+        FORMATION_COLUMN,
+        FORMATION_DOUBLE_LINE_2,
+        FORMATION_DOUBLE_LINE_1,
+        FORMATION_MOP_UP
+    }
+};
+
+static const int LAYOUT_BUTTON_INDEXES_OTHER[2][FORMATIONS_PER_LEGION] = {
+    {
+        FORMATION_SINGLE_LINE_1,
+        FORMATION_SINGLE_LINE_2,
+        FORMATION_DOUBLE_LINE_1,
+        FORMATION_DOUBLE_LINE_2,
+        FORMATION_MOP_UP
+    },
+    {
+        FORMATION_SINGLE_LINE_2,
+        FORMATION_SINGLE_LINE_1,
+        FORMATION_DOUBLE_LINE_2,
+        FORMATION_DOUBLE_LINE_1,
+        FORMATION_MOP_UP
+    }
+};
+
+static void column_formation_button(int x, int y, int selected)
```

Comment:
All the `<xxx>_formation_button` functions are effectively drawing functions.

Since they're only used for the military sidebar I've kept them in this file.

However, should they go to `src/widget/formation_layout_buttons.c`?